coderd/database/migrations/000313_preset_prebuilds.up.sql

```diff
@@ -0,0 +1,6 @@
+ALTER TABLE template_version_presets
+    ADD COLUMN desired_instances     INT NULL,
+    ADD COLUMN invalidate_after_secs INT NULL DEFAULT 0;
+
+-- We should not be able to have presets with the same name for a particular template version.
+CREATE UNIQUE INDEX idx_unique_preset_name ON template_version_presets (name, template_version_id);
```

Comment:
@SasSwart this index make sense, but I want to double check, can it be the case that this UNIQUE constraint already broken by some of out customers? and this migration will fail in this case?

Reply:
Yeah, it could fail in that case. Might need to write a query that renames any violations to `originalname_num` or something.